metriq_gym/qplatform/job.py

```diff
@@ -33,3 +34,74 @@ def _(quantum_job: BraketQuantumTask) -> float:
     return (
         quantum_job._task.metadata()["endedAt"] - quantum_job._task.metadata()["createdAt"]
     ).total_seconds()
+
+@dataclass
+class JobStatusInfo:
+    """Provider agnostic job status information."""
+
+    status: str
+    queue_position: int | None = None
+
+
+@singledispatch
+def job_status(quantum_job: QuantumJob) -> JobStatusInfo:
+    """Return the provider specific status and queue position if available."""
+    try:
+        status_obj = quantum_job.status()
+        status = getattr(status_obj, "name", str(status_obj))
+    except Exception:
+        status = "UNKNOWN"
+    queue_position = None
+    for attr in ["queue_position", "queue_info", "queue_position_retrieval"]:
+        if hasattr(quantum_job, attr):
+            try:
+                info = getattr(quantum_job, attr)
+                info = info() if callable(info) else info
+                if hasattr(info, "position"):
+                    info = info.position
+                if info is not None:
+                    queue_position = int(info)
+                break
+            except Exception:
+                pass
+    return JobStatusInfo(status=status, queue_position=queue_position)
+
+
+@job_status.register
+def _(quantum_job: QiskitJob) -> JobStatusInfo:
+    status = getattr(quantum_job._job.status(), "name", str(quantum_job._job.status()))
```

Comment:
It looks like those subclasses already have `status()` implemented in qBraid, e.g. for QiskitJob: https://github.com/qBraid/qBraid/blob/d3f6614b5ea3c58fe0fe1a0a79be68638673e26c/qbraid/runtime/ibm/job.py#L96
Since that's the case, we don't want to introspect into the nested job in metriq-gym.